mlir/lib/Bindings/Python/Rewrite.cpp

```diff
@@ -36,6 +41,43 @@ class PyPDLPatternModule {
   }
   MlirPDLPatternModule get() { return module; }
 
+  static nb::object fromPDLValue(MlirPDLValue value) {
+    if (mlirPDLValueIsValue(value)) {
+      return nb::cast(mlirPDLValueAsValue(value));
+    }
+    if (mlirPDLValueIsOperation(value)) {
+      return nb::cast(mlirPDLValueAsOperation(value));
+    }
+    if (mlirPDLValueIsAttribute(value)) {
+      return nb::cast(mlirPDLValueAsAttribute(value));
+    }
+    if (mlirPDLValueIsType(value)) {
+      return nb::cast(mlirPDLValueAsType(value));
+    }
+
+    throw std::runtime_error("unsupported PDL value type");
+  }
+
+  void registerRewriteFunction(const std::string &name,
+                               const nb::callable &fn) {
```

Comment:
this is also `static` actually right? can be moved out like the other one.

Reply:
Yeah we need `this` here.